### PR TITLE
Add [test] and update CONTRIBUTING

### PR DESCRIPTION
**Pull request by dlwh**
Makes it easier to set up dev env.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Now make your changes. Make sure to include additional tests if necessary.
 Next verify the tests all pass:
 
 ```bash
-pip install pytest
+pip install -e .[test]
```
Comment (patrick-kidger) [inline CONTRIBUTING.md:37]: I think this will need quote marks to work in most shells.

File: pyproject.toml [hunk 2]
```diff
@@ -55,3 +55,10 @@ reportIncompatibleMethodOverride = true
 reportIncompatibleVariableOverride = false  # Incompatible with eqx.AbstractVar
 reportFunctionMemberAccess = false
 include = ["equinox", "tests"]
+
+[project.optional-dependencies]
+test = [
```
Comment (patrick-kidger) [inline pyproject.toml:60]: Nit: can we call this `tests` instead?

**Review by patrick-kidger — COMMENTED** [review]
This LGTM! I've been meaning to convert this over for a while. Thank you for tweaking this :)

Could you also (a) deleted `tests/requirements.txt` and (b) update https://github.com/patrick-kidger/equinox/blob/main/.github/workflows/run_tests.yml appropriately?

**Comment by dlwh** [comment]
PTAL

**Comment by patrick-kidger** [comment]
LGTM! Thanks for the improvement :)

**Comment by dlwh** [comment]
Of course, I should probably do it in my own projects...

**Comment by patrick-kidger** [comment]
Unfortunately I had to revert this one. The problem arises with https://github.com/patrick-kidger/equinox/blob/5a40d2ed61871b400674ff3b3c6e8f5ba410d899/.github/workflows/release.yml#L19 that we don't want to install Equinox itself here. That's already being done for us as a source or wheel distribution. And in fact we *can't* install Equinox using `pip install .`, as e.g. its `pyproject.toml` file may not even be present in the wheel. (In practice the first error was Hatch complaining about the lack of `LICENSE` file.)

Ah well.

**Pull request merged**